oauthenticator/globus.py

```diff
@@ -248,33 +251,69 @@ async def get_users_groups_ids(self, tokens):
 
         return user_group_ids
 
-    async def user_is_authorized(self, auth_model):
+    # TODO: Mark this as breaking change because before globus_admin_groups and globus_user_groups
+    #       were the ones that were being considered to determine the admin and access status of a user
+    #       and not their union with the generic admin_users and allowed_users like it is happening now.
+    async def check_allowed(self, username, auth_model):
+        """
+        Returns True for users allowed to be authorized.
+
+        Overrides the OAuthenticator.check_allowed implementation to allow users
+        either part of `allowed_users` or `allowed_organizations`, and not just those
+        part of `allowed_users`.
+        """
+        # allow admin users recognized via admin_users or update_auth_model
+        if auth_model["admin"]:
+            return True
+
+        # FIXME: consider overriding the `is_admin`` and add this logic there
         tokens = self.get_globus_tokens(auth_model["auth_state"]["token_response"])
+        user_group_ids = await self.get_users_groups_ids(tokens)
+        if self.admin_globus_groups:
+            if self.check_user_in_groups(user_group_ids, self.admin_globus_groups):
+                auth_model["admin"] = True
+                return True
```

Comment:
I commented in google.py that I think we should set admin status in `update_auth_model`, and not in `is_admin` or `check_allowed`. Check out that comment!